File: BB.Core/Model/Comment.cs
using System;
using System.Collections.Generic;

namespace BB.Core.Model
{
    public class Comment
    {
        public Comment()
        {
            UserReactions = new List<UserReaction>();
            ChildComments = new List<Comment>();
        }

        public Guid CommentId { get; set; }

        public string Text { get; set; }
        
        public DateTime DatePosted { get; set; }

        public Guid? ParentCommentId { get; set; }

        public long UserID { get; set; }


        public virtual User User { get; set; }

        public virtual Comment ParentComment { get; set; }

        public virtual ICollection<UserReaction> UserReactions { get; set; }

        public virtual ICollection<Comment> ChildComments { get; set; }
    }
}

File: BB.Web/app/Model/Comment/Comment.ts
import u = require("../Users/User");
import User = u.User;
import p = require("../Products/Product");
import Product = p.Product;
import ur = require("../Users/UserReaction");
import UserReaction = ur.UserReaction;

export class Comment {
    public CommentId: string;
    public Text: string;
    public DataPosted: Date;
    public ParentCommentId?: string;
    public UserID: number;
    public ProductId?: number;
    public User: User;
    public ParentComment: Comment;
    public Product: Product;
    public UserReactions:UserReaction[];
    public ChildComments: Comment[]; 

}